CMakeLists.txt

```diff
@@ -0,0 +1,278 @@
+  src/documentmanager.cpp
+  src/documentstatistics.cpp
+  src/documentstatisticswidget.cpp
+  src/exportdialog.cpp
+  src/exporter.cpp
+  src/exporterfactory.cpp
+  src/exportformat.cpp
+  src/htmlpreview.cpp
+  src/localedialog.cpp
+  src/mainwindow.cpp
+  src/markdowndocument.cpp
+  src/markdowneditor.cpp
+  src/markdownhighlighter.cpp
+  src/markdownast.cpp
+  src/markdownnode.cpp
+  src/memoryarena.cpp
+  src/messageboxhelper.cpp
+  src/outlinewidget.cpp
+  src/preferencesdialog.cpp
+  src/previewoptionsdialog.cpp
+  src/sandboxedwebpage.cpp
+  src/sessionstatistics.cpp
+  src/sessionstatisticswidget.cpp
+  src/sidebar.cpp
+  src/simplefontdialog.cpp
+  src/stringobserver.cpp
+  src/stylesheetbuilder.cpp
+  src/textblockdata.cpp
+  src/theme.cpp
+  src/themeeditordialog.cpp
+  src/themerepository.cpp
+  src/themeselectiondialog.cpp
+  src/timelabel.cpp
+  src/color_button.cpp
+  src/findreplace.cpp
+  src/spelling/dictionary_manager.cpp
+  src/spelling/spell_checker.cpp
+)
+
+set(ghostwriter_HEADERS
+  src/abstractstatisticswidget.h
+  src/appsettings.h
+  src/cmarkgfmapi.h
+  src/cmarkgfmexporter.h
+  src/colorscheme.h
+  src/colorschemepreviewer.h
+  src/commandlineexporter.h
+  src/documenthistory.h
+  src/documentmanager.h
+  src/documentstatistics.h
+  src/documentstatisticswidget.h
+  src/exportdialog.h
+  src/exporter.h
+  src/exporterfactory.h
+  src/exportformat.h
+  src/htmlpreview.h
+  src/localedialog.h
+  src/mainwindow.h
+  src/markdowndocument.h
+  src/markdowneditor.h
+  src/markdowneditortypes.h
+  src/markdownhighlighter.h
+  src/markdownast.h
+  src/markdownnode.h
+  src/markdownstates.h
+  src/memoryarena.h
+  src/messageboxhelper.h
+  src/outlinewidget.h
+  src/preferencesdialog.h
+  src/previewoptionsdialog.h
+  src/sandboxedwebpage.h
+  src/sessionstatistics.h
+  src/sessionstatisticswidget.h
+  src/sidebar.h
+  src/simplefontdialog.h
+  src/stringobserver.h
+  src/stylesheetbuilder.h
+  src/textblockdata.h
+  src/theme.h
+  src/themeeditordialog.h
+  src/themerepository.h
+  src/themeselectiondialog.h
+  src/timelabel.h
+  src/findreplace.h
+  src/color_button.h
+  src/spelling/abstract_dictionary.h
+  src/spelling/abstract_dictionary_provider.h
+  src/spelling/dictionary_manager.h
+  src/spelling/dictionary_ref.h
+  src/spelling/spell_checker.h
+)
+
+add_executable(ghostwriter
+  ${ghostwriter_SOURCES}
+  resources.qrc
+)
+
+target_include_directories(ghostwriter PRIVATE
+  ${CMAKE_CURRENT_SOURCE_DIR}/src
+  ${CMAKE_SOURCE_DIR}
+)
+
+set_target_properties(ghostwriter PROPERTIES
+  AUTOMOC TRUE
+  AUTORCC TRUE
+)
+
+target_compile_features(ghostwriter PRIVATE
+  cxx_std_11
+  # XXX: Plus any specific features the code uses from this list:
+  # https://cmake.org/cmake/help/latest/prop_gbl/CMAKE_CXX_KNOWN_FEATURES.html
+)
+
+# Compile warnings
+target_compile_options(ghostwriter PRIVATE
+  $<$<OR:$<CXX_COMPILER_ID:Clang>,$<CXX_COMPILER_ID:AppleClang>,$<CXX_COMPILER_ID:GNU>>:-Wall>
+  $<$<CXX_COMPILER_ID:MSVC>:/W4>
+)
```

Comment:
I noticed the `.pro` file had `CONFIG += warn_on` in it, so I added this. It could be made conditional via a cmake `option()`, though. Even if it defaults to `ON`, it would at least give builders a way to disable warnings if desired.

That'd look like:
```cmake
option(ENABLE_WARNINGS "Build with warnings enabled" ON)
if(ENABLE_WARNINGS)
  target_compile_options(ghostwriter PRIVATE
    $<$<OR:$<CXX_COMPILER_ID:Clang>,$<CXX_COMPILER_ID:AppleClang>,$<CXX_COMPILER_ID:GNU>>:-Wall>
    $<$<CXX_COMPILER_ID:MSVC>:/W4>
  )
endif()
```
And then turning warnings off would just be a matter of adding `-DENABLE_WARNINGS:BOOL=OFF` to the `cmake` command line.